lib/iris/tests/unit/common/metadata/test_CubeMetadata.py

```diff
@@ -247,6 +426,28 @@ def test_op_different__attribute_value(self, op_leniency):
             assert not lmetadata.__eq__(rmetadata)
             assert not rmetadata.__eq__(lmetadata)
 
+    @pytest.mark.parametrize("testcase", _ATTRS_TESTCASE_NAMES)
+    def test_op__splitattributes_cases(self, op_leniency, testcase):
+        # Check results for various global/local values of the same attribute.
+        # N.B. 'cases' dict specifies the expected results for each testcase.
+        check_splitattrs_op(
+            check_testcase=testcase,
+            check_lenient=op_leniency == "lenient",
+            op="equal",
+            cases={
+                "same": ("GaLb:GaLb", [True]),
+                "extra_global": ("GaL-:G-L-", [False, True]),
+                "extra_local": ("G-La:G-L-", [False, True]),
+                "same_global_local": ("GaL-:G-La", [False, True]),
+                "diff_global_local": ("GaL-:G-Lb", [False, True]),
+                "diffglobal_nolocal": ("GaL-:GbL-", [False]),
+                "diffglobal_samelocal": ("GaLc:GbLc", [False]),
+                "difflocal_noglobal": ("G-La:G-Lb", [False]),
+                "difflocal_sameglobal": ("GaLc:GaLd", [False]),
```

Comment:
How did you choose which circumstances to cover?

For instance there is nothing for same-global, one-local (with one local being absent).

I can't tell if these are oversights, or deliberate decisions based on the mechanics.

Reply:
You're right, but it's triggered me to review all of this.
I'm now confident that more cases are needed, but also there are more factors I haven't accounted for.

Having prototyped a more comprehensive approach to that, I came up with a full matrix with 1152 tests (!!)
So I realised that I need to test AFAP for the expected 'regularities' in behaviour. Thus :
  * local and global attributes of a given attribute(name) should be calculated exactly the same way
  * local and global values (for same attribute) should be calculated independently of each other
  * each result for "left OP right" should be identical to "right OP left"

However, the basic calculation (combining a right + left value), must still be tested for each of the 3 operations, both strict+lenient, so there's still a fair number of tests.

I think I'm nearly on top of this now,
but I'm building a new testing mechanism, so don't fret about that code for now.

Reply:
I have now removed the idea of "targetted" testcases with selected combinations of factors.
The new approach is , broadly :  [more matrixing](https://github.com/SciTools/iris/pull/5538/files#diff-cac8162ceb18e4c84f7a5fcd4b382197c2b42841aa402fd4bc0e545925b6b218R165-R166), leverage the expected ['same' results over different factors](https://github.com/SciTools/iris/pull/5538/files#diff-cac8162ceb18e4c84f7a5fcd4b382197c2b42841aa402fd4bc0e545925b6b218R145-R147), and [test the "local+global independence" separately from the other factors](https://github.com/SciTools/iris/pull/5538/files#diff-cac8162ceb18e4c84f7a5fcd4b382197c2b42841aa402fd4bc0e545925b6b218R403-R425), to keep the number of tests under control.